src/main/java/org/primefaces/extensions/renderer/CommandButtonSingleClickRenderer.java

```diff
@@ -57,13 +54,17 @@ protected String getToggleJS(final String widgetVar, final boolean enabled) {
         return String.format("var w=PF('%s');if(w){w.%sable();};", widgetVar, enabled ? "en" : "dis");
     }
 
-    protected String originalAttributeValue(final CommandButton button, final String current, final String key) {
-        String value = (String) button.getAttributes().get(key);
-        if (value != null) {
-            return value;
+    protected String getAttributeValue(final FacesContext context, final CommandButton button, final String attribute) {
+        String key = "CommandButtonSingleClickRenderer:" + attribute;
+        ValueExpression ve = (ValueExpression) button.getAttributes().get(key);
+        if (ve == null) {
+            ve = button.getValueExpression(attribute);
+        }
+        if (ve != null) {
+            button.getAttributes().put(key, ve);
```

Comment:
Don't think we need to store the expression. `button.getValueExpression(attribute)` should do.

Reply:
Did you test this out?  I was thinking the same thing that we need to store off the expression before it gets evaluated.

Reply:
No, didn't test this yet. I was going to.

Reply:
Yeah, we don't need to store them.